File: MPP-Java/src/main/java/domain/validators/ExcursionValidator.java
package domain.validators;

import domain.Excursion;

public class ExcursionValidator implements Validator<Excursion> {
    @Override
    public void validate(Excursion entity) throws ValidationException {
        String errors = "";
        if(entity.getPrice() <= 0){
            errors += "Price must be greater than 0!\n";
        }
        if(entity.getSeats() <= 0){
            errors += "Number of seats must be greater than 0!\n";
        }
        if(entity.getCompany().equals("")) {
            errors += "Company name is null!\n";
        }
        if(!errors.equals("")){
            throw new ValidationException(errors);
        }
    }
}

File: C_Sharp/C_Sharp/domain/validators/ExcursionValidator.cs
namespace C_Sharp.domain
{
    public class ExcursionValidator : IValidator<Excursion>
    {
        public void validate(Excursion entity)
        {
            string errors = "";
            if(entity.Price <= 0){
                errors += "Price must be greater than 0!\n";
            }
            if(entity.Seats <= 0){
                errors += "Number of seats must be greater than 0!\n";
            }
            if(entity.Company == "") {
                errors += "Company name is null!\n";
            }
            if(errors != ""){
                throw new ValidationException(errors);
            }
        }
    }
}